easybuild/tools/filetools.py

```diff
@@ -250,69 +249,13 @@ def det_common_path_prefix(paths):
         return None
 
 
-def download_file(filename, url, path):
```

Comment:
I'm in favour of the new module you introduced, but don't just move functions somewhere else with deprecating their current location.

reintroduce `download_file` here as follows:

```python
from easybuild.tools.download import download_file as _download_file
...
def download_file(*args, **kwargs):
    """DEPRECATED: Replaced by download_file in easybuild.tools.download"""
    _log.deprecated("The download_file function has moved to easybuild.tools.download", '3.0')
    return _download_file(*args, **kwargs)
```